server/src/com/cloud/network/vpc/VpcManagerImpl.java

```diff
@@ -747,6 +810,67 @@ public VpcOffering updateVpcOffering(final long vpcOffId, final String vpcOfferi
     }
 
     @Override
+    public CIDR[] getZoneSuperCidrList(final long zoneId) throws BadCIDRException {
+        final String superCIDRList = _dcDao.getDetail(zoneId, Params.SUPER_CIDR.name());
+        if (superCIDRList != null) {
+            String[] str_cidr_list = superCIDRList.split(",");
+            CIDR[] cidr_list = NetUtils.convertToCIDR(str_cidr_list);
+            if (!NetUtils.cidrListConsistency(cidr_list)) {
+                throw new InvalidParameterValueException("The cidr list is not consistent " + Arrays.toString(cidr_list));
+            }
+            return cidr_list;
+        } else {
+            throw new InvalidParameterValueException("Zone super cidr is null check cloud.data_center_details for zone id=" + zoneId);
+        }
+    }
+
+    private List<CIDR> getAllVpcCidrs() throws BadCIDRException {
+        // get all the used cidrs
+        List<CIDR> usedSubnets = new ArrayList<CIDR>();
+        for (Vpc vpc : _vpcDao.listAll()) {
+            if (_vpcSrvcDao.areServicesSupportedInVpc(vpc.getId(), Service.VPCDynamicRouting)) {
+                usedSubnets.add(CIDRFactory.getCIDR(vpc.getCidr()));
+            }
+        }
+        return usedSubnets;
+    }
+
+    @Override
+    @ActionEvent(eventType = EventTypes.EVENT_VPC_CREATE, eventDescription = "creating vpc", create = true)
+    public synchronized Vpc createDynamicVpc(final long zoneId, final long vpcOffId, final long vpcOwnerId, final String vpcName, final String displayText, final String netmask,
+            String networkDomain, final Boolean displayVpc) throws ResourceAllocationException, BadCIDRException {
+        if (netmask != null) {
+            if (!NetUtils.isValidNetmask(netmask)) {
```

Comment:
Please consider implementing this validation on the as a static method on the ``Netmask`` interface for IPv4/IPv6 dual stack support.